src/hugo/content/en-us/implementations/aws/device_bootstrap/aws-iot-certificate-vending-machine/_index.md

```diff
@@ -0,0 +1,425 @@
+
+Your `device_token` is calculated by hashing functions using your secret salt string. You need to implement some extra layers to increase the protection of registration logic and protect your sensitive material. 
+- If your device's computing environment provides hardware-based encryption blocks or read-protected memory blocks, you can leverage these features to reduce the possible attack surface. Almost all modern MCU's provide these feature sets out of the box, otherwise, you may need to include a TPM (Trusted Platform Module) component to your hardware design.
+- You can use different secrets salts for different batches/series of devices. This makes only one batch/series of devices use the same secret durinng the registration. This reduces the blast radius of leakage of your secret salt.
+
+### Service Quotas
+This implementation uses an AWS Lambda function to make AWS IoT Core API calls. You need to identify possible batch registration events and estimated throughput of IoT API calls to avoid hitting service limits and quotas.
+
+### Logging and Monitoring of Device Registration
+
+Implement a logging and monitoring mechanism for critical steps and KPI's regarding your device registration use-case to build a registration dashboard. This monitoring strategy can include the following:
+
+- CloudWatch log insights of the CVM Lambda function
+- Lambda function invocation metrics
+- IoT Service CloudWatch metrics
+- IoT Core service logs
+- IoT Core service events
+
+### API Response Times, Errors, and Retries
+
+Your IoT device's registration implementation should be aware of the CVM API response times and possible error codes. In case of an error, it should trigger retries or reporting scenarios within a well-defined flow.
+
+CVM API response time depends on possible Lambda cold starts, IoT Core API calls run in the lambda function, and your other custom logic in the function.
+
+Some possible application-level error catches are implemented in the lambda function as examples. But they may differ according to your use case.
+
+### Multi-Region Strategy
+This implementation shows to provision IoT resources on a single defined region. Also, the device database on Amazon DynamoDB and Amazon API Gateway endpoint are deployed on a single region. If your use case requires the creation of IoT resources in different regions, you can implement additional logic in the Lambda function and call APIs in different regions to create resources. If your usecase requires complete isolation of resources in different regions or a multi-region fault tolerance, you can deploy CVM solution to multiple resources. You can leverage multi-region features of services used, e.g. **global tables** feature of Amazon DynamoDB which provides fully managed, multi-region, and multi-active database, replicates automatically across your choice of AWS Regions.
```

Comment:
shows _how_ to
_in_ a single region.

Check the entire document and replace _on_ with _in_ or _within_. 

check for _usecase_ and replace with _use-case_,

in different regions or a multi-region fault tolerance -> in different regions, or a multi-region fault tolerance (added comma)

you can deploy CVM solution to multiple resources -> you can deploy _the_ CVM solution to multiple _regions_

e.g. global tables feature of Amazon DynamoDB , (add comma)

and multi-active database, _that_ replicates automatically across your choice of AWS Regions.

Reply:
All fixed.